Celeste.Mod.mm/Mod/Everest/Everest.Loader.cs

```diff
@@ -598,33 +598,62 @@ public static class Loader {
                             ConstructorInfo ctor;
                             MethodInfo gen;
 
+
                             gen = type.GetMethod(genName, new Type[] { typeof(Level), typeof(LevelData), typeof(Vector2), typeof(EntityData) });
                             if (gen != null && gen.IsStatic && gen.ReturnType.IsCompatible(typeof(Entity))) {
-                                loader = (level, levelData, offset, entityData) => (Entity) gen.Invoke(null, new object[] { level, levelData, offset, entityData });
+                                loader = (level, levelData, offset, entityData) => {
+                                    Entity e = (Entity) gen.Invoke(null, new object[] { level, levelData, offset, entityData });
+                                    (e as patch_Entity).EntityData = entityData;
+                                    return e;
+                                };
                                 goto RegisterEntityLoader;
```

Comment:
This check is not guaranteeable in this specific case. I'll add a note in comments in its stead though.
Situationally, this arises because you can have, say,
```cs
[CustomEntity("ExampleHelper/ExampleEntity = Load")]
public class ExampleEntity1 : Solid { 
    public static void Load(Level level, LevelData levelData, Vector2 offset, EntityData entityData) => new ExampleEntity2(entityData, offset);
    ...
}

public class ExampleEntity2 { ... }
```

Reply:
right, I misread that a bit, oops